CoreConversationView/src/au/gov/asd/tac/constellation/views/conversationview/EnhancedTextArea.java

```diff
@@ -0,0 +1,137 @@
+        this.setEditable(false);
+        this.setPadding(insets);
+        this.appendText(text);
+
+        // Implementation for the 'Copy' context menu option.
+        final MenuItem copyTextMenuItem = new MenuItem("Copy");
+        copyTextMenuItem.setOnAction(event -> {
+            final StringSelection ss = new StringSelection(this.getSelectedText());
+            final Clipboard clipboard = Toolkit.getDefaultToolkit().getSystemClipboard();
+            clipboard.setContents(ss, ConstellationClipboardOwner.getOwner());
+        });
+
+        // Implementation for the 'Select All' context menu option.
+        final MenuItem selectAllTextMenuItem = new MenuItem("Select All");
+        selectAllTextMenuItem.setOnAction(event -> this.selectAll());
+
+        // Implementation for the 'Select on Graph' context menu option.
+        final MenuItem selectOnGraphMenuItem = new MenuItem("Select on Graph");
+        selectOnGraphMenuItem.setOnAction(event -> {
+            final BitSet elementIds = new BitSet();
+            elementIds.set(contribution.getMessage().getTransaction());
+
+            PluginExecution.withPlugin(VisualGraphPluginRegistry.CHANGE_SELECTION)
+                    .withParameter(ChangeSelectionPlugin.ELEMENT_BIT_SET_PARAMETER_ID, elementIds)
+                    .withParameter(ChangeSelectionPlugin.ELEMENT_TYPE_PARAMETER_ID, new ElementTypeParameterValue(GraphElementType.TRANSACTION))
+                    .withParameter(ChangeSelectionPlugin.SELECTION_MODE_PARAMETER_ID, SelectionMode.REPLACE)
+                    .executeLater(GraphManager.getDefault().getActiveGraph());
+        });
+
+        final ContextMenu contextMenu = new ContextMenu();
+        contextMenu.getItems().add(copyTextMenuItem);
+        contextMenu.getItems().add(selectAllTextMenuItem);
+        contextMenu.getItems().add(selectOnGraphMenuItem);
+
+        this.setOnContextMenuRequested(event -> {
+            contextMenu.show(this, event.getScreenX(), event.getScreenY());
+            copyTextMenuItem.setDisable(this.getSelectedText().isEmpty());
+        });
+    }
+
+    /**
+     * Finds and highlights passages of the given string found within the text
+     * of the EnhancedTextArea and returns the count of passages found.
+     *
+     * @param searchText Text passage to be searched for.
+     * @return Count of passages of searched text found.
+     */
+    public int highlightText(final String searchText) {
+
+        // Clear any previous highlighting.
+        this.setStyle(0, this.getText().length(), "-rtfx-background-color: transparent;");
+
+        List<Tuple<Integer, Integer>> found = new ArrayList<>();
+
+        if (!searchText.isEmpty()) {
+            found = StringUtilities.searchRange(this.getText(), searchText);
+
+            if (!found.isEmpty()) {
+                found.forEach(location -> this.setStyle(location.getFirst(), location.getSecond(), "-rtfx-background-color: yellow;"));
+            }
```

Comment:
I could be wrong but if `found` is empty then the statement inside the if block just won't do anything so I think you can take the `found.forEach()` out of the if block (and remove the if)

Reply:
So that code inside the if highlights the found letters/passages, whose indexes are store in the array called 'found'. If its empty then there's nothing to highlight so no need to go into the if.

Reply:
Nevermind, works the same if the code isn't surrounded by an 'if'.